src/muz/spacer/spacer_context.cpp

```diff
@@ -3321,6 +3347,96 @@ void context::predecessor_eh()
     }
 }
 
+// Update a partial model to be consistent over reachable facts in pt
+
+// Conceptually, a reachable fact is asserted as tag == > fact,
+// where tag is a Boolean literal and fact a formula. When the model is partial,
+// it is possible that tag is true, but model.eval(fact) is unknown
+// This function fixes the model by flipping the value of tag in
+// this case.
+bool pred_transformer::mk_mdl_rf_consistent(const datalog::rule *r,
+                                            model *model) {
+  expr_ref rf(m);
+  reach_fact_ref_vector child_reach_facts;
+
+  SASSERT(r != nullptr);
+  ptr_vector<func_decl> preds;
+  find_predecessors(*r, preds);
+  for (unsigned i = 0; i < preds.size(); i++) {
+    func_decl *pred = preds[i];
+    bool atleast_one_true = false;
+    pred_transformer &ch_pt = ctx.get_pred_transformer(pred);
+    // get all reach facts of pred used in the model
+    expr_ref o_ch_reach(m);
+    reach_fact_ref_vector used_rfs;
+    ch_pt.get_all_used_rf(*model, i, used_rfs);
+    for (auto *rf : used_rfs) {
+      pm.formula_n2o(rf->get(), o_ch_reach, i);
+      if (!model->is_true(o_ch_reach)) {
+        func_decl *tag = to_app(rf->tag())->get_decl();
+        set_true_in_mdl(model, tag);
+      } else
+        atleast_one_true = true;
+    }
+    if (used_rfs.size() > 0 && !atleast_one_true) {
+      TRACE("spacer_detail",
+            tout << "model does not satisfy any reachable fact\n";);
+      return false;
+    }
+  }
+  return true;
+}
+
+// Update a partial model to be consistent over reachable facts.
+
+// Conceptually, a reachable fact is asserted as tag == > fact,
+// where tag is a Boolean literal and fact a formula. When the model is partial,
+// it is possible that tag is true, but model.eval(fact) is unknown
+// This function fixes the model by flipping the value of tag in
+// this case.
+bool context::mk_mdl_rf_consistent(model *mdl) {
+  reach_fact_ref_vector used_rfs;
+  expr_ref exp(m);
+  for (auto &rel : m_rels) {
+    bool atleast_one_true = false;
+    pred_transformer &pt = *rel.m_value;
+    used_rfs.reset();
+    pt.get_all_used_rf(*mdl, used_rfs);
+    for (auto *rf : used_rfs) {
+      if (!mdl->is_true(rf->get())) {
+        func_decl *tag = to_app(rf->tag())->get_decl();
+        set_true_in_mdl(mdl, tag);
+      } else
+        atleast_one_true = true;
+    }
+    if (used_rfs.size() > 0 && !atleast_one_true) {
+      TRACE("spacer_detail",
+            tout << "model does not satisfy any reachable fact\n";);
+      return false;
+    }
+  }
+  return true;
+}
+
+// Handle cases where solver returns unknown but returns a good enough model
+// model is good enough if it satisfies
+// 1. all the reachable states whose tag is set in the model
+// 2. Tr && pob
+lbool context::handle_unknown(pob &n, const datalog::rule *r, model_ref model) {
+  if (r == nullptr) {
```

Comment:
4 spaces
https://www.bing.com/videos/search?q=tabs+vs+spaces&&view=detail&mid=D8125B650297896DCD76D8125B650297896DCD76&&FORM=VRDGAR

Reply:
But .... I used clang format.

Reply:
it is configurable